### PR TITLE
web: drop `bootstrap` depenedency

### LINKED ISSUE #29200: Get rid of Bootstrap style imports
## Problem statement

We depend on Bootstrap styles in our global styles a lot. But we don't plan to move forward with Bootstrap because we're working on our design system called Wildcard. To ease the transition to our styles and gain more control over visual changes, we want to move Bootstrap styles _that we need_ into our codebase and eventually drop Bootstrap dependency.

## Success criteria

All Bootstrap CSS rules that we need are incorporated in our codebase. 

## Linked issues

- Bootstrap removal
  - #29252
  - #29253
  - #29355
  - #29356
  - #30279
  - #30280
  - #30281
  - #31574
  - #32438
  - #32439
  - #32458
  - #32819
  - #32820
  - #37324
  - #37325
  - #37914
  - #40293
  - #40294
  - #40295
  - #40296
- Post removal clean up
  - #41397
  - #41398
  - #41399

## Implementation details

Let's start with these SCSS files and create a separate PR for each one:
1. `client/branded/src/global-styles/list-group.scss`
2. `client/branded/src/global-styles/nav.scss`

Notes:
1. We want to migrate away from SCSS variables to CSS variables when we bring Bootstrap styles to our codebase.
    - Some Bootstrap SCSS variables redefined on our side are located here: `client/branded/src/global-styles/index.scss`. Keep that in mind while migrating to CSS variables. 
2. We don't need to bring all styles defined in the imported Bootstrap package. Check what styles do we use in the codebase first.

## Time estimate

- Pull requests with ~450 lines changed should take **6 hours** at maximum. Ping the reviewer in the spec pull request if time-consuming changes are required.
- Split the work into multiple pull requests if the total diff is bigger than 450 lines of code.

/cc @taylorsperry @muratsu

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

This PR removes `bootstrap` from `package.json` dependencies and updates documentation to reflect that we do not use `bootstrap` directly anymore.

Closes #29200

## Test plan

1. Check out the updated documentation.
2. Ensure that CI is green.
3. Ensure that `bootstrap` is not present in the lock file.

## App preview:

- [Web](https://sg-web-vb-drop-bootstrap-dependency.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-iuqeffhxxu.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
🔥

**Review by umpox — APPROVED** [review]
🚀

File: client/branded/src/global-styles/GlobalStylesStory/GlobalStyles.story.tsx [hunk 1]
```diff
@@ -253,136 +215,6 @@ export const Layout: Story = () => (
     </>
 )
 
-export const ButtonGroups: Story = () => {
```
Comment (courier-new) [inline client/branded/src/global-styles/GlobalStylesStory/GlobalStyles.story.tsx:256]: I know "button groups" was a Bootstrap concept, but we're keeping our Wildcard `ButtonGroup` implementation, right? I see we have a separate [Storybook story](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/wildcard/src/components/Button/story/Button.story.tsx?L130%3A1=) for `ButtonGroup`s already, but the usages here don't look like they overlap with those ones. Is it worth porting any of these to that story?

Comment (valerybugakov) [reply client/branded/src/global-styles/GlobalStylesStory/GlobalStyles.story.tsx:256]: @courier-new, good catch! Fixing it [here](https://github.com/sourcegraph/sourcegraph/pull/41487), please take a look!

**Pull request merged**